aries-site/src/examples/templates/dashboards/content/chartCards/RulesAudit.js

```diff
@@ -0,0 +1,191 @@
+import { Fragment, useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { Box, Grid, Meter, Text } from 'grommet';
+import { Calendar } from 'grommet-icons';
+import { ChartCard, Measure, MeterGroup } from '../../components';
+import { defaultWindow, REPORT_WINDOW_MAP } from './utils';
+
+const mockData = require('../../../../../data/mockData/compliance.json');
+
+export const RulesAudit = ({ period }) => {
+  const { audits, checks } = mockData;
+  const [reportWindow, setReportWindow] = useState(defaultWindow);
+  const [completedAudits, setCompletedAudits] = useState(null);
+  const [nextAudit, setNextAudit] = useState(null);
+  const [rules, setRules] = useState([]);
+  const [frameworks, setFrameworks] = useState([]);
+  const [auditResults, setAuditResults] = useState([]);
+
+  // Set reporting window
+  useEffect(() => {
+    setReportWindow(prevWindow => ({
+      ...prevWindow,
+      begin: new Date(
+        new Date().setDate(
+          prevWindow.end.getDate() - REPORT_WINDOW_MAP[period],
+        ),
+      ),
+    }));
+  }, [period]);
+
+  // Isolate completed vs. upcoming audits/
+  useEffect(() => {
+    const completed = audits
+      .filter(audit => audit.completedDate)
+      .sort((a, b) => new Date(b.completedDate) - new Date(a.completedDate))
+      .filter(
+        audit =>
+          new Date(audit.completedDate) < reportWindow.end &&
+          new Date(audit.completedDate) > reportWindow.begin,
+      );
+    const next = audits
+      .sort((a, b) => new Date(a.scheduledDate) - new Date(b.scheduledDate))
+      .find(audit => new Date(audit.scheduledDate) > reportWindow.end);
+    setCompletedAudits(completed);
+    setNextAudit(next);
+  }, [audits, reportWindow.begin, reportWindow.end]);
+
+  // Get results from completed audits.
+  useEffect(() => {
+    if (completedAudits) {
+      const nextRules = checks.filter(check =>
+        completedAudits.map(audit => audit.id).includes(check.auditId),
+      );
+      const nextFrameworks = [];
+      checks.forEach(check => {
+        if (!nextFrameworks.includes(check.frameworkId)) {
+          nextFrameworks.push(check.frameworkId);
+        }
+      });
+      setRules(nextRules);
+      setFrameworks(nextFrameworks);
+    }
+  }, [checks, completedAudits]);
+
+  // Assemble chart data
+  useEffect(() => {
+    const nextAuditResults = [
+      { result: 'pass', count: 0, percent: 0, color: 'status-ok' },
```

Comment:
can we make the `pass` `fail` and `error` capital? for the label on the chart

Reply:
Done.

Reply:
Clarifying that the capitalization takes place at render, not in the data.